### PR TITLE
Add disk monitoring

### PR DESCRIPTION
**Pull request by perllaghu**
Adds disk monitoring.

I've also done a couple of tweaks - like moving the UI labelling to indicators to their individual View file, and use these labels as the main reference.

(I didn't try and get the react code to take them from the schema file, but maybe a future task)

### CODE REVIEW

File: packages/labextension/src/widget.tsx [hunk 1]
```diff
@@ -296,7 +296,7 @@ const KernelUsage = (props: {
                 {props.trans.__('Process ID:')} {usage.pid}
               </div>
               <div className="jp-KernelUsage-separator">
-                {props.trans.__('CPU:')} {usage.kernel_cpu}
+                {props.trans.__('HPU:')} {usage.kernel_cpu}
```
Comment (krassowski) [inline packages/labextension/src/widget.tsx:299]: ?

Comment (perllaghu) [reply packages/labextension/src/widget.tsx:299]: Oops....my bad - I spent ages trying to figure out what which bit of code changed which bit of UI...

Once you have a full set of changes, I'll push a complete set

Comment (perllaghu) [reply packages/labextension/src/widget.tsx]: Done

**Comment by perllaghu** [comment]
I am unclear as to what is failing the linting... everything passes on my workstation

File: jupyter_resource_usage/api.py [hunk 2]
```diff
@@ -75,6 +75,20 @@ async def get(self):
...
+            try:
+                disk_info = psutil.disk_usage(config.disk_path)
+            except:
+                pass
+            else:
+                metrics.update(disk_used=disk_info.used, disk_total=disk_info.total)
+                limits["disk"] = {"disk": disk_info.total}
+                if config.disk_warning_threshold != 0:
+                    limits["disk"]["warn"] = (disk_info.total - disk_info.used) < (
+                        disk_info.total * config.cpu_warning_threshold
```
Comment (iandesj) [inline jupyter_resource_usage/api.py:89]: shouldn't this be 
```python
disk_info.total * config.disk_warning_threshold
```
??

**Review by iandesj — CHANGES_REQUESTED** [review]
I've added a comment for what might be a potential logic error when calculating disk warning state.

File: jupyter_resource_usage/api.py (hunk 2 above)
Comment (perllaghu) [reply jupyter_resource_usage/api.py:89]: Indeed it should!

Comment (iandesj) [reply jupyter_resource_usage/api.py:89]: Additionally, the warning state calculated is never handled. It looks like the current client-side code handles warning state for memory only. I'm playing with this right now to show the disk warning!

Comment (perllaghu) [reply jupyter_resource_usage/api.py]: More than happy for you to contribute....
Question: does the CPU have a parallel warning state that should be handled?

Comment (iandesj) [reply jupyter_resource_usage/api.py]: Great question! I'm more than happy to contribute to your branch and close PR #233, which I've pulled in your changes to and did the React work for the warnings. I was in a flow state this AM and wasn't sure how active this PR was going to be, so I threw those changes together.

Check it out and let me know what you think!

**Comment by perllaghu** [comment]
This PR has been superseded by https://github.com/jupyter-server/jupyter-resource-usage/pull/233

**Pull request closed without merging**